src/psyclone/psyir/nodes/operation.py

```diff
@@ -211,7 +211,7 @@ class BinaryOperation(Operation):
         # Array Query Operators
         'SIZE', 'LBOUND', 'UBOUND',
         # Matrix and Vector Operators
-        'MATMUL'
+        'MATMUL', 'DOT_PRODUCT'
         ])
     _non_elemental_ops = [Operator.SUM, Operator.MATMUL, Operator.SIZE,
```

Comment:
I think DOT_PRODUCT needs to be included in the list of _non_elemental_ops as it returns a single scalar given an array input.

Reply:
Good spot.